Analysis/IsolatedLeptonFinder/src/IsolatedLeptonFinderProcessor.cc

```diff
@@ -242,31 +254,63 @@ IsolatedLeptonFinderProcessor::IsolatedLeptonFinderProcessor()
 				"Maximum Z in jet-based isolation",
 				_jetIsoVetoMaxZ,
 				float(0.6));
+
+		registerProcessorParameter( "WhichLeptons",
+				"Use DRESSED, UNDRESSED or BOTH lepton algorithms",
+				_whichLeptons,
+				std::string("UNDRESSED"));
+
+		registerProcessorParameter( "MergeCloseElectrons",
+				"Merge close-by electrons into higher energy lepton",
+				_mergeCloseElectrons,
+				bool(false));
+
+		registerProcessorParameter( "DressPhotonConeAngle",
+				"Half-angle (in degrees) of the cone used for lepton dressing with photons",
+				_dressPhotonConeAngle,
+				float(1));
+
+		registerProcessorParameter( "MergeLeptonConeAngle",
+				"Half-angle (in degrees) of the cone used for lepton merging",
+				_mergeLeptonConeAngle,
+				float(2));
+
+		registerProcessorParameter( "UsePandoraIDs",
+				"Use Pandora particle IDs for algorithm",
+				_usePandoraIDs,
+				bool(false));
 	}
 
 
-void IsolatedLeptonFinderProcessor::init() { 
+void IsolatedLeptonFinderProcessor::init() {
 	streamlog_out(DEBUG) << "   init called  " << std::endl ;
 	printParameters() ;
 }
 
-void IsolatedLeptonFinderProcessor::processRunHeader( LCRunHeader* run) { 
-} 
+void IsolatedLeptonFinderProcessor::processEvent( LCEvent * evt ) {
 
-void IsolatedLeptonFinderProcessor::processEvent( LCEvent * evt ) { 
+	streamlog_out(MESSAGE) <<std::endl;
+	streamlog_out(MESSAGE) << "processing event: " << evt->getEventNumber() << "   in run:  " << evt->getRunNumber() << std::endl ;
 
+	_dressedPFOs.clear();
 	_rpJetMap.clear();
 
 	_pfoCol = evt->getCollection( _inputPFOsCollection ) ;
 
-	// Output PFOs removed isolated leptons 
+	// Output PFOs removed isolated leptons
 	LCCollectionVec* otPFOsRemovedIsoLepCol = new LCCollectionVec( LCIO::RECONSTRUCTEDPARTICLE ) ;
 	otPFOsRemovedIsoLepCol->setSubset(true) ;
 
 	// Output PFOs of isolated leptons
 	LCCollectionVec* otIsoLepCol = new LCCollectionVec( LCIO::RECONSTRUCTEDPARTICLE );
 	otIsoLepCol->setSubset(true);
 
+	// Output PFOs removed dressed isolated leptons
+	LCCollectionVec* otPFOsRemovedDressedIsoLepCol = new LCCollectionVec( LCIO::RECONSTRUCTEDPARTICLE ) ;
```

Comment:
As it is optional which collections are stored you need to cleanup. Easiest with unique_ptr

```c++
atuo* otPFOsRemovedDressedIsoLepCol = std::unique_ptr<LCCollectionVec>(new LCCollectionVec( LCIO::RECONSTRUCTEDPARTICLE );
```
and then later
```c++
evt->addCollection(otPFOsRemovedDressedIsoLepCol.release(), "collectionName");
```
If the collection is not added it gets cleaned up automatically


PS: Would be nice to have c++14 and make_unique

Reply:
Hi Andre,
your suggestion yields the following error:
```
error: unable to deduce ‘auto*’ from ‘std::unique_ptr<IMPL::LCCollectionVec>((std::__cxx11::basic_string<char>(EVENT::LCIO::RECONSTRUCTEDPARTICLE, std::allocator<char>()), (operator new(256ul), (<statement>, ((IMPL::LCCollectionVec*)<anonymous>)))))’
  auto* otPFOsRemovedDressedIsoLepCol = std::unique_ptr<LCCollectionVec>(new LCCollectionVec( LCIO::RECONSTRUCTEDPARTICLE ) );

```

Reply:
Done! The `auto*` -> `auto` was the solution